apprise/plugins/NotifyNtfy.py

```diff
@@ -0,0 +1,264 @@
+
+    # The default descriptive name associated with the Notification
+    service_name = 'Ntfy'
+
+    # The services URL
+    service_url = 'https://ntfy.sh/'
+
+    # Insecure protocol (for those self hosted requests)
+    protocol = 'ntfy'
+
+    # The default protocol
+    secure_protocol = 'ntfys'
+
+    # A URL that takes you to the setup/help of the specific protocol
+    setup_url = 'https://github.com/caronc/apprise/wiki/Notify_ntfy'
+
+    # Ntfy uses the https protocol
+    notify_url = 'https://ntfy.sh/{topic}'
+
+    # Message time to live (if remote client isn't around to receive it)
+    time_to_live = 2419200
+
+    # Define object templates
+    templates = (
+        '{schema}://{host}/{topic}',
+    )
+
+    # Define our template tokens
+    template_tokens = dict(NotifyBase.template_tokens, **{
+        'topic': {
+            'name': _('Topic'),
+            'type': 'string',
+            'required': False,
+        },
+        'host': {
+            'name': _('Hostname'),
+            'type': 'string',
+            'required': True,
+        },
+        'tags': {
+            'name': _('Tags'),
+            'type': 'string',
+            'required': False,
+        }
+    })
+
+    # Define our template arguments
+    template_args = dict(NotifyBase.template_args, **{
+        'priority': {
+            'name': _('Priority'),
+            'type': 'choice:string',
+            'values': NTFY_PRIORITIES,
+            'default': NtfyPriority.DEFAULT,
+        },
+    })
+
+    def __init__(self, topic, priority=None, tags=None, **kwargs):
+        """
+        Initialize Ntfy Object
+        """
+        super(NotifyNtfy, self).__init__(**kwargs)
+        if not self.topic:
+            msg = 'A topic name must be provided.'
+            self.logger.warning(msg)
+            raise TypeError(msg)
+
+        # The priority of the message
+        if priority not in NTFY_PRIORITIES:
+            self.priority = self.template_args['priority']['default']
+        else:
+            self.priority = priority
+
+        # Any optional tags to attach to the notification
+        self.tags = tags or ''
+
+        # prepare our fullpath
+        self.fullpath = kwargs.get('fullpath')
+        if not isinstance(self.fullpath, six.string_types):
+            self.fullpath = '/'
+
+    def send(self, body, title='', notify_type=NotifyType.INFO, **kwargs):
+        """
+        Perform Ntfy Notification
+        """
+
+        # error tracking (used for function return)
+        has_error = False
+
+        # Prepare our headers
+        headers = {}
+        if self.priority != NtfyPriority.DEFAULT:
+            headers['X-Priority'] = NtfyPriority.get_value(self.priority)
+        if title:
+            headers['X-Title'] = title
+        if self.tags:
+            headers['X-Tags'] = self.tags
+
+        # Prepare our payload
+        payload = body
+
+        # Prepare our Ntfy URL
+        url = self.notify_url.format(topic=self.topic)
```

Comment:
This seems like self-hosted instances won't be supported, because the `notify_url` is `https://ntfy.sh/{topic}`, right?

Reply:
Correct. That's one of the things I have to come back to. Just wanted to get it working to start.